src/Function.cpp

```diff
@@ -574,6 +635,12 @@ Function::OutputForwardDecl(std::ostream &out)
 		int power = rnd_upto(16);
 		out << " __attribute__((aligned(" << (1 << power) << ")))";
 	}
+	vector<FunctionAttribute*>::iterator itr;
+	for(itr = attributes.begin(); itr < attributes.end(); itr++)
+		(*itr)->OutputAttributes(out);
+	if(CGOptions::func_attr_flag())
```

Comment:
Shouldn't this be `if (attr_emitted)` ?

Reply:
> Shouldn't this be `if (attr_emitted)` ?

Here, `CGOptions::func_attr_flag()` sets only when command line is set for generating common function attributes. I can use `attr_emitted` but using `CGOptions::func_attr_flag()` is appropriate here.

Reply:
Both approaches are fine. If you want to check `attr_emitted`, I can change to it.

Reply:
That's not correct. You can have enabled `CGOptions::func_attr_flag()`, but you'll be not lucky and no attribute will be emitted with `FunctionAttribute::OutputAttribute`. Then you'll emit here `))` which will lead to a syntax error.

Reply:
My bad. You are right. I have changed it to `attr_emitted`

Reply:
Great.